crates/testenv/src/lib.rs

```diff
@@ -186,18 +186,45 @@ impl TestEnv {
         Ok((bt.height as usize, block.block_hash()))
     }
 
-    /// This method waits for the Electrum notification indicating that a new block has been mined.
-    /// `timeout` is the maximum [`Duration`] we want to wait for a response from Electrsd.
-    pub fn wait_until_electrum_sees_block(&self, timeout: Duration) -> anyhow::Result<()> {
-        self.electrsd.client.block_headers_subscribe()?;
+    /// Wait until Electrum is aware of a block of a given `block_height` (and optionally, matches
+    /// the given `block_hash`).
+    pub fn wait_until_electrum_sees_block(
+        &self,
+        block_height: usize,
+        block_hash: Option<BlockHash>,
+        timeout: Duration,
+    ) -> anyhow::Result<()> {
+        self.electrsd.trigger()?;
+        // NOTE: There is a reason why we use the subscribe endpoint specifically. You may think
+        // just polling Electrs via Electrum for a block header at height `block_height` and
+        // checking whether `block_hash` matches is enough. However, having the header polling call
+        // up to date does not necessarily mean the spk histories are up to date. On the other hand,
+        // getting a notification for a new block tip does mean that the confirmed spk histories
+        // are up to date up to and including the new notified tip. This is all due to the internal
+        // workings of Electrs.
```

Comment:
I may need to word this better.

Reply:
How about:

```
// NOTE: We use the subscribe endpoint because polling Electrs for a block header at 
// `block_height` and verifying the `block_hash` does not ensure that the spk histories 
// are current. In contrast, getting a notification for a new block tip ensures that the
// confirmed spk histories are current, including the new notified tip. This is a result of
// the internal workings of Electrs.
```